packages/types/src/type/ExtrinsicEra.ts

```
@@ -2,33 +2,152 @@
+   */
+  get mortalEra (): MortalEra {
+    return this.value as MortalEra;
+  }
+
+  /**
+   * @description Encodes the value as a Uint8Array as per the parity-codec specifications
+   * @param isBare true when the value has none of the type-specific prefixes (internal)
+   */
+  toU8a (isBare?: boolean): Uint8Array {
+    if (this.index === 0 ) {
+      return super.toU8a()
+    } else {
+      return this.mortalEra.toU8a(isBare);
+    }
+  }
+}
+
+export class ImmortalEra extends U8a {
+  constructor(value?: AnyU8a) {
+    super(value);
+  }
+}
 
-import U8a from '../codec/U8a';
+export type MortalEraValue = [U8, U8];
 
 /**
- * @name ExtrinsicEra
+ * @name MortalEra
  * @description
- * The era for an extrinsic, indicating either a mortal or immortal extrinsic
+ * The MortalEra for an extrinsic, indicating period and phase
  */
-export default class ExtrinsicEra extends U8a {
-  constructor (value?: AnyU8a) {
-    super(
-      ExtrinsicEra.decodeExtrinsicEra(value)
-    );
+export class MortalEra extends Tuple {
+  constructor (value?: any) {
+    super({
+      period : U8, phase : U8
+    }, MortalEra.decodeMortalEra(value));
+  }
+
+  private static decodeMortalEra (value: EraMethod | Uint8Array | string): MortalEraValue {
+    if (isHex(value)) {
+      return MortalEra.decodeMortalEra(hexToU8a(value.toString()));
+    } else if (isU8a(value)) {
+      const u8a = value;
+      const first = u8a.subarray(0, 1);
+      let second = u8a.subarray(1, 2);
+      const encoded = new U64(new U64(first).toNumber() + (new U64(second).toNumber() << 8));
+      const period = new U64(2 << (encoded.toNumber() % (1 << 4)));
+      const factor = 12;
+      const quantizeFactor = Math.max(period.toNumber() >> factor, 1);
+      let phase = (encoded.toNumber() >> 4) * quantizeFactor;
+      if (period.toNumber() >= 4 && phase < period.toNumber()) {
+        return [new U8(period), new U8(phase)];
+      }
+      return [new U8(), new U8()];
```

there's a default return, so this line can be removed.

---

in which case the code will reach this line. maybe we should throw an error?